### ISSUE TITLE
Looping SpriteSheetAnimations incorrectly go to a Completed state if the elapsed game time is greater than AnimationDuration

### ISSUE THREAD

**Issue by DavidFidge** [craftworkgames/MonoGame.Extended#803]
In my game I have short two framed animations which I want to loop indefinitely.  Sometimes the time between updates are greater than the total animation duration * 2.  In these situations, line 65 of SpriteSheetAnimation does not subtract enough time off of CurrentTime and thus the animation incorrectly gets set to Completed and the animations stop working.

Below is a basic unit test to demonstrate the issue.  The last two theories (seconds of 4 and 5) fail because they result in CurrentTime still being >= AnimationDuration and thus IsComplete gets set to true and the animations stop playing.

```
        [Theory]
        [InlineData(0, 0)]
        [InlineData(0, 0.9f)]
        [InlineData(1, 1f)]
        [InlineData(1, 1.9f)]
        [InlineData(0, 2f)]
        [InlineData(0, 2.9f)]
        [InlineData(1, 3f)]
        [InlineData(0, 4f)]
        [InlineData(1, 5f)]
        public void Looping_SpriteSheetAnimation_Should_Return_Correct_Frame_And_Not_Complete(int expectedTextureRegionIndex, float time)
        {
            var textureRegion2D1 = new TextureRegion2D("Region 1", null, new Rectangle());
            var textureRegion2D2 = new TextureRegion2D("Region 2", null, new Rectangle());

            var textureRegions = new[] { textureRegion2D1, textureRegion2D2 };

            var spriteSheetAnimationData = new SpriteSheetAnimationData(
                new[] { 0, 1 },
                1,
                true,
                false
            );

            var spriteSheetAnimation = new SpriteSheetAnimation("Test", textureRegions, spriteSheetAnimationData);

            spriteSheetAnimation.Play();
            var gameTime = new GameTime(TimeSpan.Zero, TimeSpan.FromSeconds(time));

            spriteSheetAnimation.Update(gameTime);

            Assert.Equal(textureRegions[expectedTextureRegionIndex], spriteSheetAnimation.CurrentFrame);
            Assert.False(spriteSheetAnimation.IsComplete);
        }
```

The issue also happens when the time between updates is a little over the Animation Duration.  In these circumstances, the CurrentTime will accumulate the residual amount of the result of CurrentTime - AnimationDuration and the animation will eventually incorrectly go to a Completed state once the residual amount causes CurrentTime -= AnimationDuration to still exceed AnimationDuration.

**Comment by DavidFidge** [comment]
The fix should be very easy - change line 65 to CurrentTime %= AnimationDuration;

**Comment by DavidFidge** [comment]
Proposed fix in pull request #806

**Comment by Gandifil** [comment]
Referenced PR has been merged.
